apps/api/src/app/inbox/dtos/get-notifications-count-request.dto.ts

```diff
@@ -1,20 +1,40 @@
-import { IsArray, IsBoolean, IsOptional, IsString } from 'class-validator';
-import { Transform } from 'class-transformer';
+import { BadRequestException } from '@nestjs/common';
+import { IsArray, IsDefined, ValidateNested, IsBoolean, IsOptional, IsString, ArrayMaxSize } from 'class-validator';
+import { plainToClass, Transform, Type } from 'class-transformer';
 import { NotificationFilter } from '../utils/types';
 
-export class GetNotificationsCountRequestDto implements NotificationFilter {
+export class NotificationsFilter implements NotificationFilter {
   @IsOptional()
   @IsArray()
   @IsString({ each: true })
   tags?: string[];
 
   @IsOptional()
   @IsBoolean()
-  @Transform(({ value }) => value === 'true')
   read?: boolean;
 
   @IsOptional()
   @IsBoolean()
-  @Transform(({ value }) => value === 'true')
   archived?: boolean;
 }
+
+export class GetNotificationsCountRequestDto {
+  @IsDefined()
+  @Transform(({ value }) => {
+    try {
+      const filters = JSON.parse(value);
+      if (Array.isArray(filters)) {
+        return filters.map((el) => plainToClass(NotificationsFilter, el));
+      }
```

Comment:
The filters are sent in the query params as the stringified JSON object. We need to parse it back from string to object and run the validation on its fields.

Reply:
The stringified JSON filters in counts creates a discrepancy between the count and the notifications endpoint where one filter is specified via separate query string parameters.

I don't have a good suggestion yet but I suggest we add a task to revise this post-release and opt-in for more alignment between the two endpoints.

Reply:
ok, I'll add one